pkg/disk/partition_table.go

```diff
@@ -678,6 +689,62 @@ func (pt *PartitionTable) ensureLVM() error {
 	return nil
 }
 
+// ensureBtrfs will ensure that the root partition is on a btrfs subvolume, i.e. if
+// it currently is not, it will wrap it in one
+func (pt *PartitionTable) ensureBtrfs() error {
+
+	rootPath := entityPath(pt, "/")
+	if rootPath == nil {
+		panic("no root mountpoint for PartitionTable")
+	}
+
+	// we need a /boot partition to boot btrfs, ensure one exists
+	bootPath := entityPath(pt, "/boot")
+	if bootPath == nil {
+		_, err := pt.CreateMountpoint("/boot", 512*common.MiB)
+
+		if err != nil {
+			return err
+		}
+
+		rootPath = entityPath(pt, "/")
+	}
+
+	parent := rootPath[1] // NB: entityPath has reversed order
+
+	if _, ok := parent.(*Btrfs); ok {
+		return nil
+	} else if part, ok := parent.(*Partition); ok {
```

Comment:
If I am not misreading we could follow effective go [0] and avoid the not strictly needed nested else here(?) (because the if returns when taken). So something like(?):
```go
@@ -655,7 +651,8 @@ func (pt *PartitionTable) ensureLVM() error {
 
        if _, ok := parent.(*LVMLogicalVolume); ok {
                return nil
-       } else if part, ok := parent.(*Partition); ok {
+       }
+       if part, ok := parent.(*Partition); ok {
                filesystem := part.Payload
 
                vg := &LVMVolumeGroup{
```

[0] https://go.dev/doc/effective_go#if